samcli/commands/deploy/command.py

```diff
@@ -287,13 +295,23 @@ def do_cli(
             config_file=config_file,
         )
         guided_context.run()
-    elif resolve_s3 and bool(s3_bucket):
-        raise DeployResolveS3AndS3SetError()
-    elif resolve_s3:
-        s3_bucket = manage_stack(profile=profile, region=region)
-        click.echo(f"\n\t\tManaged S3 bucket: {s3_bucket}")
-        click.echo("\t\tA different default S3 bucket can be set in samconfig.toml")
-        click.echo("\t\tOr by specifying --s3-bucket explicitly.")
+    else:
+        if resolve_s3:
+            if bool(s3_bucket):
+                raise DeployResolveS3AndS3SetError()
+            s3_bucket = manage_stack(profile=profile, region=region)
+            click.echo(f"\n\t\tManaged S3 bucket: {s3_bucket}")
+            click.echo("\t\tA different default S3 bucket can be set in samconfig.toml")
+            click.echo("\t\tOr by specifying --s3-bucket explicitly.")
+
+        if resolve_image_repos:
+            if image_repositories is None:
+                image_repositories = {}
+            manager_helper = CompanionStackManagerHelper(
```

Comment:
Feels weird to instantiate a `CompanionStackManagerHelper` to only use its `manager` functions.
Can't we instantiate a `CompanionStackManager` instead?
Or could we merge `CompanionStackManagerHelper` into `CompanionStackManager`?


(Following the [discussion with Sriram](https://github.com/aws/aws-sam-cli/pull/2815/files#r620576568))

Reply:
`CompanionStackManagerHelper` is used here to gather function logical IDs and resolve cases where some functions have image repos already specified.
As for merging with `CompanionStackManager`, the helper class was created with the intention to not bloat `CompanionStackManager` and separate out functionalities that are not directly related to managing the stack.